platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -660,7 +671,7 @@ public long findKey(final K key) throws IOException {
     @Override
     public Hash loadHash(final long path) throws IOException {
         if (path < 0) {
-            throw new IllegalArgumentException("path is less than 0");
+            throw new IllegalArgumentException("Path (" + path + ") is not valid");
```

Comment:
Minor: it's a bit confusing when one can call `updateValidRange(-1, -1)`, supposedly declaring `-1` valid, and then still throw an exception on such a value.

I'm not suggesting to change this because `-1` path truly looks invalid, and so the logic in the loadHash() method here seems correct. But I'm just pointing out a confusing part of the current implementation which may be difficult to grasp.

Reply:
I see your point... `updateValidRange(-1, -1)` actually means "no valid paths" rather than "-1 is a valid path". If paths were Longs, not longs, `null`s would do the trick, but there would be different issue.

On the other hand, methods like this are a part of `VirtualDataSource` interface. At that level, path -1 is invalid by any means, no doubts about that. `updateValidRange()` only exists at MerkleDb level, it's implementation details how long list index classes store their range. So I don't see a big problem here.

Reply:
Let me add a short comment to `VirtualDataSource.loadLeafRecord(long)` JavaDoc:

```
If the path is outside the current data source's leaf path range, this method returns {@code null}.
```

A similar note is in `loadHash()`